### DISCUSSION TITLE
API changes before official v1.0

### DISCUSSION THREAD

**Discussion by talentedmrjones** [cue-sh/stax#23] [Ideas]
thinking about some major changes to `stax`:
- merge all instances to a single instance which would also require us to :arrow_heading_down: 
- remove path information from log, or potentially get it via `Value.Pos().Filename()`
- remove `--include` and `--exclude`. This same functionality can be accomplished with globbing 
- remove `~/.stax/config.stx.cue` and use only the local config co-located with cue.mod
- move outputs back to `cue.mod/usr`

Also wanting to conform more to the [aws sdk stack input type](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation#CreateStackInput)
- move "opinions" such as `Environment` to `Template: Metadata` or `Stack: Tags` or a single field not mapped to CreateStackInput such as `#Stack: Params?` which exists, see next time 
- `#Stack: Params` is easily confused with Parameters , rename to `#Stack: Variables`
- remove the `--environment` flag and filter based on a path value regex match: `--value Template.Metadata.Environment=dev|stage` 
this would solve the multiple stack definition issue, eliminate the need to iterate over build instances, and simplify the code and dx

**Comment by jlongtine** [comment]
Sent this over in Discord, but thought I should share here, too. 

I like this. I prefer metadata over tags for “opinions”, personally, because I doubt all of those should be propagated to all resources. It also gives us more flexibility with the structure of those opinions.

When you say “merge all instances”, do you mean from the perspective of the current folder? So you get all the stacks defined down into the leaves from your current spot, and up the stack linearly?

It’s possible that the new Value stuff changes those semantics a bit, but I’m not sure.

I'll share more thoughts soon.
